### PR TITLE
EVG-13763: Put limits on the max duration for AWS calls

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-13763

I set the max backoff to 10 seconds, in `utility.Retry` it defaults to `min * 2^maxAttempts` which in this case would be 1024 secs.

### CODE REVIEW

File: cloud/ec2_client.go [hunk 1]
```diff
@@ -155,6 +155,7 @@ type generateDeviceNameOptions struct {
 const (
 	awsClientImplRetries     = 10
 	awsClientImplStartPeriod = time.Second
+	awsClientMaxBackoff      = 10 * time.Second
```
Comment (Kimchelly) [inline cloud/ec2_client.go:158]: Can we make this higher and reduce the number of retries as well? If it's used for every single call, the exponential backoff means that it'll try for 1, 2, 4, 8 seconds and then it won't be exponential backoff beyond 4 retries. They might rate-limit us if we don't back off enough.

**Comment by Kimchelly** [comment]
Can we also put a more aggressive context timeout specifically on `GetInstanceStatus()`/`GetInstanceStatuses()` in `set-cloud-hosts-ready`? I'm primarily concerned with that one above all the others taking a long time since the job is a bottleneck.

**Comment by bsamek** [comment]
> Can we also put a more aggressive context timeout specifically on `GetInstanceStatus()`/`GetInstanceStatuses()` in `set-cloud-hosts-ready`? I'm primarily concerned with that one above all the others taking a long time since the job is a bottleneck.

Right. Should we really reduce the retry duration for other calls? If we're asking for a fleet or on-demand instance, it's presumably fine for the job to take a long time, since each job corresponds to only a single resource. The problem seems specific to set-cloud-hosts-ready, since the batch operation seems to mean that one slow resource can slow the others.

I think we should only change the behavior of set-cloud-hosts-ready, not the behavior of other jobs.

File: cloud/ec2_client.go (hunk 1 above)
Comment (julianedwards) [reply cloud/ec2_client.go:158]: Should we just get rid of the max backoff then and reduce the number of retries?

**Review by Kimchelly — APPROVED** [review]

File: cloud/ec2_client.go (hunk 1 above)
Comment (Kimchelly) [reply cloud/ec2_client.go]: We resolved this by just reducing the number of retries and not imposing a global per-attempt timeout on every single call.

**Pull request merged**